### PR TITLE
Update ./gradlew check to run on test files

### PR DESCRIPTION
**Pull request by AustinShalit**
- Continuous integration services now run the check task
- Adds the [taskTree plugin](https://plugins.gradle.org/plugin/com.dorongold.task-tree) to view tasks that will be executed (utility plugin)

Check runs these tasks now (this is the tree from core):
```
:core:check
+--- :core:checkstyleGenerated
+--- :core:checkstyleMain
+--- :core:checkstyleTest
+--- :core:findbugsMain
|    \--- :core:classes
|         +--- :core:compileJava
|         |    \--- :core:generateCodeFromSource
|         \--- :core:processResources
+--- :core:findbugsTest
|    +--- :core:classes
|    |    +--- :core:compileJava
|    |    |    \--- :core:generateCodeFromSource
|    |    \--- :core:processResources
|    \--- :core:testClasses
|         +--- :core:compileTestJava
|         |    \--- :core:classes
|         |         +--- :core:compileJava
|         |         |    \--- :core:generateCodeFromSource
|         |         \--- :core:processResources
|         \--- :core:processTestResources
+--- :core:pmdMain
+--- :core:pmdTest
|    \--- :core:classes
|         +--- :core:compileJava
|         |    \--- :core:generateCodeFromSource
|         \--- :core:processResources
\--- :core:test
     +--- :core:classes
     |    +--- :core:compileJava
     |    |    \--- :core:generateCodeFromSource
     |    \--- :core:processResources
     \--- :core:testClasses
          +--- :core:compileTestJava
          |    \--- :core:classes
          |         +--- :core:compileJava
          |         |    \--- :core:generateCodeFromSource
          |         \--- :core:processResources
          \--- :core:processTestResources
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ install:
   - '[ "${TRAVIS_PULL_REQUEST}" = "false" ] && ./gradlew :ui:jfxNative --stacktrace || ./gradlew --stacktrace '
 
 script:
-  - ./gradlew checkstyleMain checkstyleTest pmdMain pmdTest findbugsMain findbugsTest jacocoTestReport jacocoRootReport test --stacktrace -Pheadless=true -PlogTests
+  - ./gradlew check jacocoTestReport jacocoRootReport --stacktrace -Pheadless=true -PlogTests
```
Comment (AustinShalit) [inline .travis.yml]: I thought about making check run the jacoco stuff also but decided against it.  It just will bloat the check task and is not 'required' for a pull request.

Comment (JLLeitschuh) [inline .travis.yml]: How much longer does it really take to run the tests with `jacoco`?

Comment (AustinShalit) [inline .travis.yml]: Almost nothing...

Comment (JLLeitschuh) [inline .travis.yml]: So is it really worth it? I think whatever "bloat" seems negligible.

Comment (AustinShalit) [inline .travis.yml]: I think that it does not make sense. Tasks that run with check should have a chance of failing.  Jacoco is a report -- not a check or test.

Comment (JLLeitschuh) [inline .travis.yml]: This is fair.
Okay, fine.

**Comment by JLLeitschuh** [comment]
LGTM!

**Pull request merged**